debezium-connector-mariadb/src/main/java/io/debezium/connector/mariadb/snapshot/lock/AtLeastOnceSnapshotLock.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.debezium.connector.mariadb.snapshot.lock;
+
+import java.util.Map;
+
+import io.debezium.connector.mariadb.MariaDbConnectorConfig;
+import io.debezium.snapshot.spi.SnapshotLock;
+
+public class AtLeastOnceSnapshotLock extends DefaultSnapshotLock implements SnapshotLock {
```

Comment:
I think we agreed on putting this mode into binlog module and make it available for both MariaDB and MySQL in case the user uses arbitrary storage module for which we cannt be sure that TX with consistent snapshot is working.